airflow-core/docs/administration-and-deployment/plugins.rst

```diff
@@ -194,27 +196,49 @@ definitions in Airflow.
+    # Creating an external view that will be rendered in the Airflow UI.
     external_view_with_metadata = {
-        "name": "Name of the Iframe View as displayed in the UI",
+        # Name of the external view, this will be displayed in the UI.
+        "name": "Name of the External View",
         # Source URL of the external view. This URL can be templated using context variables, depending on the location where the external view is rendered
         # the context variables available will be different, i.e a subset of (DAG_ID, RUN_ID, TASK_ID, MAP_INDEX).
-        "href": "https://example.com/{DAG_ID}/{RUN_ID}/{TASK_ID}",
-        # Destination of the iframe view. This is used to determine where the iframe will be loaded in the UI.
+        "href": "https://example.com/{DAG_ID}/{RUN_ID}/{TASK_ID}/{MAP_INDEX}",
+        # Destination of the external view. This is used to determine where the view will be loaded in the UI.
         # Supported locations are Literal["nav", "dag", "dag_run", "task", "task_instance"], default to "nav".
         "destination": "dag_run",
         # Optional icon, url to an svg file.
         "icon": "https://example.com/icon.svg",
         # Optional dark icon for the dark theme, url to an svg file. If not provided, "icon" will be used for both light and dark themes.
         "icon_dark_mode": "https://example.com/dark_icon.svg",
-        # Optional parameters, relative URL location for the iframe rendering. If not provided, external view will be rendeded as an external link. Should
-        # not contain a leading slash.
+        # Optional parameters, relative URL location for the External View rendering. If not provided, external view will be rendeded as an external link. If provided
+        # will rendered inside an Iframe in the UI. Should not contain a leading slash.
         "url_route": "my_iframe_view",
         # Optional category, only relevant for destination "nav". This is used to group the external links in the navigation bar.  We will match the existing
         # menus of ["browse", "docs", "admin", "user"] and if there's no match then create a new menu.
         "category": "browse",
     }
 
+    react_app_with_metadata = {
+        # Name of the React app, this will be displayed in the UI.
+        "name": "Name of the React App",
+        # Bundle URL of the React app. This is the URL where the React app is served from. It can be a static file or a CDN.
+        # This URL can be templated using context variables, depending on the location where the external view is rendered
+        # the context variables available will be different, i.e a subset of (DAG_ID, RUN_ID, TASK_ID, MAP_INDEX).
+        "bundle_url": "https://example.com/static/js/my_react_app.js",
+        # Destination of the iframe view. This is used to determine where the iframe will be loaded in the UI.
+        # Supported locations are Literal["nav", "dag", "dag_run", "task", "task_instance"], default to "nav".
```

Comment:
```suggestion
        # Supported locations are Literal["nav", "dag", "dag_run", "task", "task_instance"], default to "nav". It can also be put inside of an existing page, the supported views are ["dashboard", "dag_overview", "task_overview"]
```